src/components/app/index.tsx

```diff
@@ -35,6 +35,28 @@ interface State {
   error?: string;
 }
 
+async function updateCompressedImage(
+  source: SourceImage,
+  encodeData: EncoderState,
+): Promise<ImageBitmap> {
+  // Special case for identity
+  if (encodeData.type === identity.type) return source.bmp;
+
+  const compressedData = await (() => {
+    switch (encodeData.type) {
+      case mozJPEG.type: return mozJPEG.encode(source.data, encodeData.options);
+      default: throw Error(`Unexpected encoder name`);
+    }
+  })();
+
+  const blob = new Blob([compressedData], {
+    type: encoderMap[encodeData.type].mimeType,
+  });
+
+  const bitmap = await createImageBitmap(blob);
+  return bitmap;
+}
```

Comment:
This was only using state to set the error message, so I moved that to `updateImage`.